### ISSUE TITLE
Sane(r) HTTP-answercodes in phone/general token answers

### ISSUE THREAD

**Issue by olmari** [TampereHacklab/mulysa#166]
Currently phone API gives 404 in both situations where device id or phone number is not found on system. Change number not found to return `403 Forbidden` so it would reflect better the actual result. I think device id could stay 404 when not actually found. This way initiating party can tell what is actually happened and react accordingly, instead not knowing what is wrong.

Similar thoughts would work on generally token-system (NFC).

**Comment by olmari** [comment]
After discussion in dev chat, it turned out that currently APi returns 403 when a phonenumber exist with member who doesn't have "tilankäyttöoikeus", totally unknown number gives 404, and unknown device id also returns 404.

My suggestion is that API eanswers 403 to all numbers that doesn't have door access as that is only relevant information to door automation.

For future improvements if really wanted to we could define some separate status codes for number found in member but no door access vs number not found at all, and/or make abilities (webhooks?) for Mulysa to report who door was opened for. Both ways could then be used for reporting to various chats that door was opened for member etc, and with more data into board member/techdata chat.

**Comment by olmari** [comment]
It is also perfectly spec-compliant to define custom HTTP status codes for the AP purposesI, all that matters is that the "class" is correct (the first number in 3-digit statuscode), this way we could very well spec API to give for example "456 member has no door access" and "457 number unknown",  and door automation side can easily be defined to understand those. so we are really not bound to "standard" status codes only

Spec: https://tools.ietf.org/html/rfc2616#section-6.1.1

```
   HTTP status codes are extensible. HTTP applications are not required
   to understand the meaning of all registered status codes, though such
   understanding is obviously desirable. However, applications MUST
   understand the class of any status code, as indicated by the first
   digit, and treat any unrecognized response as being equivalent to the
   x00 status code of that class, with the exception that an
   unrecognized response MUST NOT be cached. For example, if an
   unrecognized status code of 431 is received by the client, it can
   safely assume that there was something wrong with its request and
   treat the response as if it had received a 400 status code. In such
   cases, user agents SHOULD present to the user the entity returned
   with the response, since that entity is likely to include human-
   readable information which will explain the unusual status.
```

and

```
   The first digit of the Status-Code defines the class of response. The
   last two digits do not have any categorization role. There are 5
   values for the first digit:

      - 1xx: Informational - Request received, continuing process

      - 2xx: Success - The action was successfully received,
        understood, and accepted

      - 3xx: Redirection - Further action must be taken in order to
        complete the request

      - 4xx: Client Error - The request contains bad syntax or cannot
        be fulfilled

      - 5xx: Server Error - The server failed to fulfill an apparently
        valid request

   The individual values of the numeric status codes defined for
   HTTP/1.1, and an example set of corresponding Reason-Phrase's, are
   presented below. The reason phrases listed here are only
   recommendations -- they MAY be replaced by local equivalents without
   affecting the protocol.
```

**Comment by olmari** [comment]
Implemented in #169
